document/RuleMetadataProfile.mdk

```diff
@@ -1 +1,165 @@
+unit of analysis as implemented by an individual tool.
+
+### `id` property { #ruleDescriptor-id }
+
+A `ruleDescriptor` object shall contain a property named `id` whose value is a string containing
+a stable, opaque identifier for the rule.
+
+~ IsoExample
+`"CA2101"`
+~
+
+### `name` property { #ruleDescriptor-name }
+
+A `ruleDescriptor` object may contain a property named `name` whose value is a string containing
+a readable rule identifier. A rule name is suitable for use as a key to other data or to
+otherwise identify a rule (e.g., within an in-source suppression) but is not guaranteed to
+be stable. In cases where the readable rule `name` contains implementation details that are
+invalidated over time, for example, a tool might alter a rule's readable name (while leaving
+the stable `id` property unchanged).
+
+~ IsoExample
+`"SpecifyMarshalingForPInvokeStringArguments"`
+~
+
+### `abstract` property { #ruleDescriptor-abstract }
+
+A `ruleDescriptor` object may contain a property named `abstract` whose value is a string containing
+a concise description of the rule. The `abstract` property should be a single sentence that is understandable
+when displayed in user interface contexts where the available space is limited to a single line of text.
+
+~ IsoExample
+`"Specify marshaling for P/Invoke string arguments"`
+~
+
+### `description` property { #ruleDescriptor-description}
+
+An `ruleDescriptor` object should contain a property named `description` whose value is a string that describes the rule.
+
+The `description` property should, as far as possible, provide details sufficient to enable resolution of issues. 
+
+The first sentence of the `description` property should provide a concise description of the rule,
+suitable for display in cases where UI is limited.
+Tools that construct `description` in this way need not provide a value for the `abstract` property.
+Tools that do not construct `abstract` in this way should provide a value for the `abstract` property,
+because otherwise, the initial portion of `description` that a viewer displays where UI space is limited
+might not be understandable.
+
+### `options` property { #ruleDescriptor-options }
+
+A `ruleDescriptor` object may contain a property named `options`, whose value is a JSON object
+consisting of an arbitrary set of string/value pairs.
+When present, the `options` objects shall describe the set of configurable options supported by the rule.
+
+The value within each string-value pair shall be a string, which may be the empty string.
+The value shall not be a JSON array or sub-object.
+
+### `formatSpecifiers` property { #ruleDescriptor-formatSpecifiers }
+
+A `ruleDescriptor` object may contain a property named `formatSpecifiers`, whose value is a JSON object
+consisting of an arbitrary set of string/value pairs.
+When present, the `formatSpecifiers` object shall describe a set of message formatting strings
+that have or may be returned by the tool.
```

Comment:
We are defining the Rule Metadata Profile here. Is it constrained to describing the message strings that happen to be specified by the set of issues in the current log file?

I think the idea is this:
* The `formatSpecifiers` object hast to include _at least_ the set of message strings sufficient to display the issues in the current log file -- that's why we included the rule metadata in the log file in the first place.
* It will probably be easier for tools to emit the entire set of messages the rule is capable of, rather than having to grovel the entire (large) set of issues to find out which subset to emit. So we should allow a superset.